base/math.jl

```diff
@@ -504,40 +504,43 @@ end
 Compute the hypotenuse ``\\sqrt{\\sum x_i^2}`` avoiding overflow and underflow.
 """
 function hypot(x::Number...)
-    # compute infnorm x (modeled on generic_vecnormMinusInf(x) in LinearAlgebra/generic.gl)
-    (v, s) = iterate(x)::Tuple
+
+    xp = promote(x...)
```

Comment:
Perhaps better to just make the signature `hypot(x::T...) where T<:Number` and then have a separate method for `hypot(x::Number...) = hypot(promote(x...))`?

Reply:
Sounds good. I'll commit a slight revision of that: `hypot(x::Number...) = hypot(promote(x...)...)` correctly passes off to `function hypot(x::T...) where T<:Number`, 

whereas `hypot(x::Number...) = hypot(promote(x...))` results in
```
julia> hypot(1.0, 0, 0)
ERROR: MethodError: no method matching hypot(::Tuple{Float64,Float64,Float64})
```
Thanks for all the help.  I'm learning a lot.

Reply:
Yes, that's what I meant :)

Reply:
With the above revisions `make test` fails with the following. I see from [NEWS.md](https://github.com/JuliaLang/julia/blob/master/NEWS.md) that this has to do with "possibly throw[ing] `UndefVarError` when accessing static parameters," but I've got no idea how to address it.

```
Error in testset ambiguous:
Test Failed at /home/gibson/gitworking/julia/test/ambiguous.jl:302
  Expression: need_to_handle_undef_sparam == Set()
   Evaluated: Set(Method[hypot(x::T...) where T<:Number in Base.Math at math.jl:510]) == Set(Any[])
ERROR: LoadError: Test run finished with errors
in expression starting at /home/gibson/gitworking/julia/test/runtests.jl:59
```

Reply:
It seems like the ambiguous case is `hypot()`. What should the hypotenuse of zero numbers be?

Reply:
`nothing`? `0.0`? Throwing an error seems most sensible to me. That's what `norm([])` does.

Reply:
In that case the fix would be to have a method that throws an error for zero arguments similar to what norm does.